sys/parsoid.js

```diff
@@ -510,23 +496,33 @@ class ParsoidService {
         });
     }
 
-    _getStashedContent(hyper, req, etag) {
+    _getStashedContent(hyper, req, tid) {
         const rp = req.params;
-        const getStash = (format) => hyper.get({
-            uri: this.getStashBucketURI(rp, format, etag.tid)
-        })
-        .then(mwUtil.decodeBody);
-
-        return P.props({
-            html: getStash('html'),
-            'data-parsoid': getStash('data-parsoid'),
-            wikitext: getStash('wikitext')
+        return hyper.get({
+            uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid)
         })
         .then((res) => {
+            res = JSON.parse(res.body.toString('utf8'));
```

Comment:
Let's be consistent and use `mwUtil.decodeBody()`

Reply:
Ulala... let's not. Judging by the history of the whole 'decodeBody' magic will have to go. It's all about zipping up the content that we don't do anymore. Plus, after split this line will go away too.